openquake/hazardlib/gsim/bchydro_2016_epistemic.py

```diff
@@ -67,12 +68,174 @@ def get_stress_factor(imt, slab=False):
     return sigma_mu / 1.65
 
 
+class FABATaperStep(object):
+    """
+    General class for a tapering function, in this case
+    a step function such that the backarc scaling term takes 0 for
+    forearc sites (negative backarc distance), and 1 for backarc sites
+    (positive backarc distance)
+    """
+    def __init__(self, **kwargs):
+        pass
+
```

Comment:
@g-weatherill I guess you are using the `**kwargs` signature here because I said that it was okay to use it in the GMPETable case. Actually it was a lesser of two evils situation: in a situation when you do not know the parameters in advance and you cannot anticipated what in the future will go in the subclasses (as it happens for subclasses of GMPETable) using `**kwargs` is acceptable. If this is the situation here I will not protest. But if you know that the parameters can be at most a, b,c, sigma, etc, it is a  lot better to be explicit and to list them in the signature. The important thing is that the signature must be the same among all subclasses, i.e. it is okay to include parameters that are not used. Unless the number of parameters is huge or it is not known.

Reply:
I'm not sure which is the better option here then. As you can see, each subclass has different inputs and even though I use the same parameter names (a, b etc.) some of them have different meanings and therefore different default values. Of course I could re-name the parameters so that each is more specific to the subclass but I'm not sure that is more elegant than using `**kwargs`. I don't anticipate the addition of more subclasses than those present here.

Reply:
Then it is easy, Graeme: leave things as they are, but add a comment explaining the choice. A good point where to add the comment is near the lines
```python
    def __init__(self, **kwargs):
        pass
```
which are a bit surprising for somebody looking at the code.